unmock/core/http.py

```diff
@@ -0,0 +1,143 @@
+from typing import Optional, List
+import http.client
+from urllib.parse import urlsplit, SplitResult
+
+from .options import UnmockOptions
+from .utils import Patchers
+
+__all__ = ["initialize", "reset"]
+
+# Backup:
+UNMOCK_AUTH = "___u__n_m_o_c_k_a_u_t__h_"
+PATCHERS = Patchers()
+STORIES = list()
+
+def parse_url(url) -> SplitResult:
+    parsed_url = urlsplit(url)
+    if parsed_url.scheme == "" or parsed_url.netloc == "":
+        return urlsplit("http://{url}".format(url=url))
+    return parsed_url
+
+def initialize(unmock_options: UnmockOptions, story: Optional[List[str]] = None, token: Optional[str] = None):
+    """Entry point to mock the standard http client. Both `urllib` and `requests` library use the
+    `http.client.HTTPConnection`, so mocking it should support their use aswell.
+
+    We mock the "low level" API (instead of the `request` method, we mock the `putrequest`, `putheader`, `endheaders`
+    and `getresponse` methods; the `request` method calls these sequentially).
+
+    HTTPSConnection also uses the regular HTTPConnection methods under the hood -> hurray!
+    """
+    global PATCHERS, STORIES
+    if story is not None:
+        STORIES += story
+
+    def unmock_putrequest(self: http.client.HTTPConnection, method, url, skip_host=False, skip_accept_encoding=False):
```

Comment:
Using `self` as function argument is a bit confusing to me, wondering if it could be given some other name... Does not really matter ofc.